tests/unit/providers/oauth-provider.spec.js

```diff
@@ -0,0 +1,452 @@
+    var provider;
+
+    beforeEach(function() {
+      angular.module('angular-oauth2.mock', [])
+        .config(function(OAuthProvider) {
+          provider = OAuthProvider;
+        });
+
+      module('angular-oauth2', 'angular-oauth2.mock');
+
+      inject(function() {});
+    });
+
+    it('should throw an error if configuration is not an object', function() {
+      try {
+        provider.configure(false);
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(TypeError);
+        e.message.should.match(/config/);
+      }
+    });
+
+    it('should throw an error if already configured', function() {
+      try {
+        provider.configure(config);
+        provider.configure(config);
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(Error);
+      }
+    });
+
+    it('should throw an error if `baseUrl` param is empty', function() {
+      try {
+        provider.configure(_.omit(config, 'baseUrl'));
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(Error);
+        e.message.should.match(/baseUrl/);
+      }
+    });
+
+    it('should remove trailing slash from `baseUrl`', function() {
+      var parsedConfig = provider.configure(_.defaults({
+        baseUrl: 'https://api.website.com/'
+      }, config));
+
+      parsedConfig.baseUrl.should.equal('https://api.website.com');
+    });
+
+    it('should throw an error if `clientId` param is empty', function() {
+      try {
+        provider.configure(_.omit(config, 'clientId'));
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(Error);
+        e.message.should.match(/clientId/);
+      }
+    });
+
+    it('should throw an error if `clientSecret` param is empty', function() {
+      try {
+        provider.configure(_.omit(config, 'clientSecret'));
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(Error);
+        e.message.should.match(/clientSecret/);
+      }
+    });
+
+    it('should throw an error if `grantPath` param is empty', function() {
+      try {
+        provider.configure(_.defaults({ grantPath: null }, config));
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(Error);
+        e.message.should.match(/grantPath/);
+      }
+    });
+
+    it('should add facing slash from `grantPath`', function() {
+      var parsedConfig = provider.configure(_.defaults({
+        grantPath: 'oauth2/token'
+      }, config));
+
+      parsedConfig.grantPath.should.equal('/oauth2/token');
+    });
+
+    it('should throw an error if `revokePath` param is empty', function() {
+      try {
+        provider.configure(_.defaults({ revokePath: null }, config));
+
+        should.fail();
+      } catch(e) {
+        e.should.be.an.instanceOf(Error);
+        e.message.should.match(/revokePath/);
+      }
+    });
+
+    it('should add facing slash from `revokePath`', function() {
```

Comment:
We need to make sure that the urls are correctly formed. For that, we remove the trailing slash from `baseUrl` and add a facing slash to `grantPath` and `revokePath`.

I'm not sure if we should be doing this and if this is the right way of doing it. Ideas?